site/zenodo_rdm/legacy/services.py

```diff
@@ -84,15 +96,60 @@ class LegacyRecordService(RDMRecordService):
     def create(self, identity, data, uow=None, expand=False):
         """Create a draft and prereserve the DOI."""
         res = super().create(identity, data, uow=uow, expand=False)
-        res = self.pids.create(
-            identity=identity,
-            id_=res.id,
-            scheme="doi",
-            expand=True,
-            uow=uow,
-        )
+        pids = data.get("pids", {})
+
+        # If pids is provided, it is created automatically.
+        # TODO confirm this is correct
+        # TODO what happens with other providers / schemes? They are allowed in RDM
+        if not pids:
+            res = self.pids.create(
+                identity=identity,
+                id_=res.id,
+                scheme="doi",
+                expand=True,
+                uow=uow,
+            )
+
+        communities = data.get("communities", [])
+        p = self.add_communities(communities, identity, uow, res._record)
```

Comment:
question: if it fails while adding communities, should we rollback all of it or ignore?

Reply:
We should be consistent with how it worked in legacy, in which case we're never failing but just ignore/skip (in case the request is already created or similar). An alternative is to handle this case here inside the service method and check before making the `add_communities` call (so that we only add the new ones). Basically do "look before you leap" instead of the usual "ask for forgiveness than permission".

In the same fashion, we should also handle removing communities :)